### PR TITLE
Transcribe use case

### PR DESCRIPTION
**Pull request by AliShaban1**

### CODE REVIEW

**Comment by AliShaban1** [comment]
I will resolve the conflicts sometime today

**Comment by AliShaban1** [comment]
Not overwriting UploadView for now for later to decide

File: src/main/java/api/WhisperTranscription.java [hunk 1]
```diff
@@ -14,11 +14,10 @@
 
 public class WhisperTranscription implements TranscriptionInterface {
 
-    final private String OPENAI_API_KEY;
+    final private String OPENAI_API_KEY="";
     final private int MAX_CHUNK_SIZE_BYTES = 20 * 1024 * 1024;
 
-    public WhisperTranscription(String apiKey) {
-        OPENAI_API_KEY = apiKey;
+    public WhisperTranscription() {
```
Comment (DrDabbidy) [inline src/main/java/api/WhisperTranscription.java:20]: I don't think we should hard-code the api key into the source, because then we can't push to github. We need to retrieve it from a file in the Main method, so I think we should revert this change.

File: src/main/java/app/.DS_Store
Comment (DrDabbidy) [inline src/main/java/app/.DS_Store:1]: We should remove these `.DS_Store` files.

File: src/main/java/app/UploadUseCaseFactory.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        import interface_adapter.ViewManagerModel;
+        import interface_adapter.transcribe.TranscribeController;
+        import interface_adapter.transcribe.TranscribePresenter;
+        import interface_adapter.transcribe.TranscribeViewModel;
+        import interface_adapter.upload.*;
+        import use_case.transcribe.TranscribeInputBoundary;
+        import use_case.transcribe.TranscribeInteractor;
+        import use_case.transcribe.TranscribeOutputBoundary;
+        import use_case.upload.*;
+        import view.UploadView;
```
Comment (DrDabbidy) [inline src/main/java/app/UploadUseCaseFactory.java:14]: Why are these indented?

File: src/main/java/entities/Episode.java [hunk 3]
```diff
@@ -20,4 +21,5 @@ public Transcript getTranscript() {
     public String getSummary() {
         return summary;
     }
+    public void setTranscript(Transcript transcript){this.transcript = transcript;}
```
Comment (DrDabbidy) [inline src/main/java/entities/Episode.java:24]: Can you reformat this?

File: src/main/java/view/UploadView.java [hunk 4]
```diff
@@ -74,8 +84,11 @@ public void propertyChange(PropertyChangeEvent evt) {
         String errorMessage = uploadState.getErrorMessage();
         if (episode != null) {
             status.setText(String.format("Successfully uploaded file for %s", episode.getTitle()));
+            transcribeController.execute(episode);
```
Comment (DrDabbidy) [inline src/main/java/view/UploadView.java:87]: Hmm, should it be here, or in the submit button action listener, after executing the upload use case?

File: src/main/java/use_case/transcribe/TranscribeInteractor.java
Comment (DrDabbidy) [inline src/main/java/use_case/transcribe/TranscribeInteractor.java:1]: Nice. Can you just remove those empty lines at the bottom?

**Review by DrDabbidy — CHANGES_REQUESTED** [review]
Logic looks mostly good. Just some formatting things and a coupl logic questions.

File: src/main/java/api/WhisperTranscription.java (hunk 1 above)
Comment (AliShaban1) [reply src/main/java/api/WhisperTranscription.java:20]: Yes we don't have to hard code it but I think we should retrieve it from a file and set it as default just so we don't need to pass it in everytime we call the api. Let me know what you think about this approach.

File: src/main/java/view/UploadView.java (hunk 4 above)
Comment (AliShaban1) [reply src/main/java/view/UploadView.java:87]: You're absolutely right it should be in the submit button action listener, I only added it here to make sure it's being executed after the upload use case is successfully executed

File: src/main/java/app/.DS_Store
Comment (AliShaban1) [reply src/main/java/app/.DS_Store:1]: Done!

File: src/main/java/app/UploadUseCaseFactory.java (hunk 2 above)
Comment (AliShaban1) [reply src/main/java/app/UploadUseCaseFactory.java]: oops thanks for catching that!

File: src/main/java/use_case/transcribe/TranscribeInteractor.java
Comment (AliShaban1) [reply src/main/java/use_case/transcribe/TranscribeInteractor.java:1]: Done!

File: src/main/java/entities/Episode.java (hunk 3 above)
Comment (AliShaban1) [reply src/main/java/entities/Episode.java]: Done

File: src/main/java/api/WhisperTranscription.java (hunk 1 above)
Comment (DrDabbidy) [reply src/main/java/api/WhisperTranscription.java:20]: We can get the api key in the `main` method and then input it when we create the object. Putting it in the class itself is not good, because it makes it specific to our uses, rather than being general.

File: src/main/java/view/UploadView.java (hunk 4 above)
Comment (DrDabbidy) [reply src/main/java/view/UploadView.java:87]: Actually, I think you're right. We only want to transcribe if the upload use case succeeds.

**Review by DrDabbidy — APPROVED** [review]
LGTM! 👍
